File: 83_RemoveDuplicatesfromSortedList/sol.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        if(head==null) return null;
        ListNode tmp = head;
        while(tmp.next!=null){
            if(tmp.val == tmp.next.val){
                tmp.next = tmp.next.next;
            }
            else{
                tmp = tmp.next;
            }
        }
        return head;
    }
}

File: 83_RemoveDuplicatesfromSortedList/sol.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def deleteDuplicates(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        tmp = head
        while tmp!=None and tmp.next!=None:
            if tmp.next.val == tmp.val:
                tmp.next = tmp.next.next
            else:
                tmp = tmp.next
        return head